rclrs/src/error.rs

```diff
@@ -352,3 +379,80 @@ impl ToResult for rcl_ret_t {
         to_rclrs_result(*self)
     }
 }
+
+/// A helper trait to disregard timeouts as not an error.
+pub trait RclrsErrorFilter {
+    /// Get the first error available, or Ok(()) if there are no errors.
+    fn first_error(self) -> Result<(), RclrsError>;
```

Comment:
That's both interesting and surprising to me. I knew that you could use collect to go from `Vec<Result<T, E>>` to `Result<Vec<T>, E>` but I didn't know `Vec<Result<(), E>>` to `Result<(), E>` was possible. This seems to be a special case for `()`. I can change the implementation of `fn first_error` to take advantage of this, although I think the code would be a bit less clear about what it's doing.

Reply:
We could also remove this function from the public API and let the users figure out what to do with a `Vec<Result<(), RclrsError>>`

Reply:
We're already using it [in the unit tests](https://github.com/ros2-rust/ros2_rust/pull/428/commits/b465f6f74eba71270e843d690e44f81ac78ca873#diff-3882c5aa1d3bc696162e78b8b0f429bbe0a2c82423da84eb012371774777c04bR449). I'd rather not replace it with `.into_iter().collect()` because I feel like `.into_iter().collect()` is much more abstruse than it needs to be for expressing something as simple as "give me the first error in this collection".